distribution/lib/Standard/Base/0.0.0-dev/src/Data/Time/Day_Of_Week.enso

```diff
@@ -0,0 +1,52 @@
+from Standard.Base import all
+
+type Day_Of_Week
+    type Sunday
+
+    type Monday
+
+    type Tuesday
+
+    type Wednesday
+
+    type Thursday
+
+    type Friday
+
+    type Saturday
+
+    ## Convert the Day_Of_Week to an Integer
+
+       Arguments:
+       - `first_day`: The first day of the week.
+       - `include_zero`: If True, first day of the week is 0 otherwise is 1.
+    to_integer : Day_Of_Week -> Boolean -> Integer
+    to_integer self first_day=Sunday start_at_zero=False =
+        day_number = case self of
+            Sunday -> 0
+            Monday -> 1
+            Tuesday -> 2
+            Wednesday -> 3
+            Thursday -> 4
+            Friday -> 5
+            Saturday -> 6
+
+        shifted = if first_day == Day_Of_Week.Sunday then day_number else
+            (day_number + 7 - (first_day.to_integer start_at_zero=True)) % 7
+
+        shifted + if start_at_zero then 0 else 1
+
+## Convert the Java int value of a DayOfWeek to a Day_Of_Week
+
+   Arguments:
+   - `value`: The first day of the week.
+from_java_value : Integer -> Day_Of_Week
+from_java_value value = case value of
+    1 -> Monday
+    2 -> Tuesday
+    3 -> Wednesday
+    4 -> Thursday
+    5 -> Friday
+    6 -> Saturday
+    7 -> Sunday
```

Comment:
When there are two functions `to_integer` and `from_integer`, I'd expect that 
```
forall Day_Of_Week x, from_integer (to_integer x) == x
```
Imagine the case when user serializes the date and then tries to deserialize it back.

Maybe add an example to the `from_java_value` method that shows how to setup `to_integer` function so that this property holds.